src/irmin-pack/unix/gc_worker.ml

```diff
@@ -0,0 +1,383 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+module Payload = Control_file.Latest_payload
+
+let buffer_size = 8192
+
+exception Pack_error = Errors.Pack_error
+
+module type Args = sig
```

Comment:
if you include `gc_intf` here, you can avoid redefining this module type, I think. maybe even move module type `S` (renamed) to `gc_intf`. For me it make sense to have gc_intf shared between gc.ml and gc_worker, but if you disagree, you can leave it as is too :)

Reply:
I'll lump my response to @Ngoguey42's [comment](https://github.com/mirage/irmin/pull/2063#discussion_r959376621) in with this.

Coming back to this with fresh eyes today, I think we ought to bring the worker functionality into `gc.ml` (as a private `Worker` module). The worker has no use outside of the `gc.ml` context, and all these little design tensions tell me that it would fit best as one module (and avoid adding another third module for sharing purposes), at least for now.